### PR TITLE
Feature/46055 total bill 1 plus2 a latest sk

### PR DESCRIPTION
**Pull request by Skumarhpa**

### CODE REVIEW

File: src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs [hunk 1]
```diff
@@ -279,6 +310,38 @@ private static CalcResultSummaryProducerDisposalFees GetProducerRow(
...
+            if (total == 0) return 0;
+
+            return Math.Round((totalLaDisposal + total2aCommsCost) / total, 8)*100;
+
+        }
+
+
+            private static decimal GetProducerPercentageCosts(decimal producerDisposalFee, decimal commsTotalBadDebt, decimal totalDebtProvision)
+        {
+            if (totalDebtProvision <= 0) return 0;
```
Comment (udaydefra) [inline src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs:341]: Should this equal to 0 or <= 0

Comment (Skumarhpa) [reply src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs]: removed this code

**Review by johnbenjamin-defra — APPROVED** [review]

File: src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs [hunk 2]
```diff
@@ -266,6 +292,11 @@ private static CalcResultSummaryProducerDisposalFees GetProducerRow(
                 ScotlandTotalwithBadDebtprovision2A = GetScotlandCommsTotal(commsCostSummary),
                 NorthernIrelandTotalwithBadDebtprovision2A = GetNorthernIrelandCommsTotal(commsCostSummary),
 
+                //section bad debt total 
+                TotalOnePlus2AFeeWithBadDebtProvision = GetTotalProducerDisposalFeeWithBadDebtProvision(materialCostSummary) + GetTotalProducerCommsFeeWithBadDebtProvision(commsCostSummary),
+                ProducerPercentageOfCosts = GetTotal1Plus2ABadDebtPercentage(GetTotalProducerDisposalFeeWithBadDebtProvision(materialCostSummary), GetTotalProducerCommsFeeWithBadDebtProvision(commsCostSummary), materials, calcResult),
```
[the remark this answers is not in the document]
Comment (Skumarhpa) [reply src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs:298]: already a method

File: src/EPR.Calculator.API.UnitTests/Builder/Summary/CalcResultSummaryBuilderTests.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Moq;
 using System.Collections.Generic;
+using System.Reflection;
```
Comment (udaydefra) [inline src/EPR.Calculator.API.UnitTests/Builder/Summary/CalcResultSummaryBuilderTests.cs:12]: Using Reflection? for?

Comment (Skumarhpa) [reply src/EPR.Calculator.API.UnitTests/Builder/Summary/CalcResultSummaryBuilderTests.cs]: unused removed.

**Review by udaydefra — APPROVED** [review]
LGTM @Skumarhpa

**Review by johnbenjamin-defra — APPROVED** [review]

**Review by sunitaguptaEviden — APPROVED** [review]

**Review by johnbenjamin-defra — DISMISSED** [review]

**Review by sunitaguptaEviden — APPROVED** [review]

**Review by johnbenjamin-defra — DISMISSED** [review]

**Review by sunitaguptaEviden — APPROVED** [review]

**Review by johnbenjamin-defra — APPROVED** [review]

**Review by sunitaguptaEviden — APPROVED** [review]

File: src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs [hunk 4]
```diff
@@ -279,6 +310,31 @@ private static CalcResultSummaryProducerDisposalFees GetProducerRow(
             };
         }
 
+        //section bad debt total
```
Comment (udaydefra) [inline src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs:313]: Please move all new functions to a new class
Also, create a subfolder for this class and call it OnePlus2A

**Review by Mazar-Shaik — APPROVED** [review]

**Review by johnbenjamin-defra — APPROVED** [review]

**Review by udaydefra — APPROVED** [review]

File: src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs (hunk 4 above)
Comment (Skumarhpa) [reply src/EPR.Calculator.API/Builder/Summary/CalcResultSummaryBuilder.cs:313]: Resolving it as per discussion in call

**Review by sunitaguptaEviden — APPROVED** [review]

**Pull request merged**